identity-did/src/document/core_document.rs

```diff
@@ -254,70 +262,104 @@ impl<T, U, V> CoreDocument<T, U, V> {
       MethodScope::VerificationRelationship(MethodRelationship::CapabilityInvocation) => {
         self.capability_invocation.append(MethodRef::Embed(method))
       }
-    }
+    };
+
+    Ok(())
   }
 
   /// Removes all references to the specified [`VerificationMethod`].
-  pub fn remove_method(&mut self, did: &CoreDIDUrl) {
-    self.authentication.remove(did);
-    self.assertion_method.remove(did);
-    self.key_agreement.remove(did);
-    self.capability_delegation.remove(did);
-    self.capability_invocation.remove(did);
-    self.verification_method.remove(did);
+  ///
+  /// # Errors
+  ///
+  /// Returns an error if the method does not exist.
+  pub fn remove_method(&mut self, did: &CoreDIDUrl) -> Result<()> {
+    let mut was_removed = false;
+    was_removed = was_removed || self.authentication.remove(did);
+    was_removed = was_removed || self.assertion_method.remove(did);
+    was_removed = was_removed || self.key_agreement.remove(did);
+    was_removed = was_removed || self.capability_delegation.remove(did);
+    was_removed = was_removed || self.capability_invocation.remove(did);
+    was_removed = was_removed || self.verification_method.remove(did);
+
+    if was_removed {
+      Ok(())
+    } else {
+      Err(Error::MethodNotFound)
+    }
   }
 
-  /// Attaches the relationship to the given method, if the method exists.
+  /// Attaches the relationship to the method resolved by `method_query`.
   ///
-  /// Note: The method needs to be in the set of verification methods,
-  /// so it cannot be an embedded one.
+  /// # Errors
+  ///
+  /// Returns an error if the method does not exist or if it is embedded.
+  /// To convert an embedded method into a generic verification method, remove it first
+  /// and insert it with [`MethodScope::VerificationMethod`].
   pub fn attach_method_relationship<'query, Q>(
     &mut self,
     method_query: Q,
     relationship: MethodRelationship,
-  ) -> Result<bool>
+  ) -> Result<()>
```

Comment:
Discussion: changing the return type from `Result<bool>` to `Result<()>` for `attach_method_relationship` and `detach_method_relationship` removes the information on whether or not those operations actually made any changes to the document (since they are idempotent). I can imagine someone using the `bool` to determine whether or not to publish a new update, for example.

A developer can still check if that relationship exists before calling `attach/detach_method_relationship` so I'm leaning towards leaving it as `Result<()>`. Any other opinions?

Reply:
Keeping it makes sense, as you say, and doesn't have any major downsides.